netbox_docker_plugin/signals.py

```diff
@@ -0,0 +1,33 @@
+""" Hooks on Django model signals. """
+
+from django.db.models.signals import pre_delete, pre_save
+from django.dispatch import receiver
+from utilities.exceptions import AbortRequest
+
+from .models import Container
+
+
+@receiver(pre_delete, sender=Container)
+def prevent_deleting_running_containers(instance, **_kwargs):
+    """ Abort deletion if container is running. """
+
+    if not instance.can_delete:
```

Comment:
```suggestion
    if instance.host.state != "deteted" and not instance.can_delete:
```

| Host.state == delete | Container.can_delete | Raise Error |
| -------------------- | -------------------- | ----------- |
| 0                    | 0                    | 1           |
| 0                    | 1                    | 0           |
| 1                    | 0                    | 0           |
| 1                    | 1                    | 0           |

Let see if the delete cascade life cylce alllow us to do that.